spec/features/warehouse_messaging_spec.rb

```diff
@@ -21,7 +21,8 @@ def testing_scenario
 
   context 'when using RabbitMQ' do
     before do
-      Rails.configuration.bunny['enabled'] = true
+      bunny_config = { 'enabled' => true, 'broker_port' => 'xxxx', 'broker_host' => 'localhost' }
+      allow(Rails.configuration).to receive(:bunny).and_return(bunny_config)
 
```

Comment:
I think this is still mutating global state, as your reloading the broker file and initialising the constant. (Which will then persist outside the test) There really shouldn't be any need to unload constants and reload files.
```
  context 'when using RabbitMQ' do
    let(:broker) { Messages::Broker.new('enabled' => true, 'broker_port' => 'xxxx', 'broker_host' => 'localhost' ) }

    before do
      stub_const("Broker::Handle", broker)
    end
```

Then replace all the cases of `allow_any_instance_of(Messages::Broker)` with `allow(broker)`.

allow_any_instance_of is mainly intended for legacy code, and its usage is discouraged.
https://relishapp.com/rspec/rspec-mocks/docs/working-with-legacy-code/any-instance

Overall though, I'm a little confused at the overall behaviour in `app/models/broker.rb` (which I realise you haven't touched). I'm not quite sure what it means by an 'empty' class definition as the whole tap/instance_eval section doesn't actually seem to do anything. Instead it looks like the whole file could be replaced by:

```ruby
module Broker
    Handle = Messages::Broker.new(Rails.configuration.bunny)
    Handle.create_connection if Rails.configuration.bunny['enabled']
end
```

Or, even better, `Broker#create_connection` could just return nil if 'enable' is false.

Reply:
I refactored as you suggested and moved this code to a new config/initializers
With this we don't need to reload the module Broker anymore, so I removed that part too so the test is more simple.